aspnetcore/includes/ForwardedHeaders.md

```diff
@@ -0,0 +1 @@
+As a general rule, Forwarded Headers Middleware should run before other middleware. This ordering ensures that the middleware relying on forwarded headers information can consume the header values for processing. Forwarded Headers Middleware can run after diagnostics and error handling middleware, but it must must be called before `UseHsts`.
```

Comment:
Calling out just UseHsts here doesn't make sense when this paragraph is included in docs that don't show the context for Startup.cs. The line from the apache and nginx docs covers this in a more general way.
"Any component that depends on the scheme, such as authentication, link generation, redirects, and geolocation, must be placed after invoking the Forwarded Headers Middleware."

That could be incorporated in here and then you could give a few examples at the end like UseHsts, UseHttpsRedirection, UseAuthentication, etc..

Reply:
Here's the original paragraph in apache/nginx

>Any component that depends on the scheme, such as authentication, link generation, redirects, and geolocation, must be placed after invoking the Forwarded Headers Middleware. As a general rule, Forwarded Headers Middleware should run before other middleware ***except diagnostics and error handling middleware. *** This ordering ensures that the middleware relying on forwarded headers information can consume the header values for processing.

***except diagnostics and error handling middleware. ***  is what needs to be fixed.

> like UseHsts, UseHttpsRedirection, UseAuthentication, etc..

`UseHttpsRedirection, UseAuthentication` are covered with the new include

As a general rule, Forwarded Headers Middleware should run before other middleware.

I'll change this to say:

~As a general rule,~ Forwarded Headers Middleware should run before other middleware.

That covers everything. I don't see the problem with

Forwarded Headers Middleware **_can_** run after diagnostics and error handling middleware, but it must must be called before `UseHsts`.

`UseHsts` is the only problematic middleware call the templates generate as it's grouped with diagnostics and error handling middleware